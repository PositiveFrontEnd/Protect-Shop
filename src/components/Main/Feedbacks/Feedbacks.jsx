import React from "react";
import DesktopFeedbacks from "./Desktop/DesktopFeedbacks";
import { useState, useEffect } from "react";
import MobileFeedbacks from "./Mobile/MobileFeedbacks";
import { useDispatch, useSelector } from "react-redux";
import { actionGetAllShopComments } from "../../../store/shopCommentsSlice";
import { selectorShopComments } from "../../../store/selectors";

const Feedbacks = () => {
  const dispatch = useDispatch();
  const shopComments = useSelector(selectorShopComments);
  const [isMobile, setIsMobile] = useState(
    window.matchMedia("(max-width: 768px)").matches
  );
  const handleMediaQuery = (event) => {
    setIsMobile(event.matches);
  };
  useEffect(() => {
    dispatch(actionGetAllShopComments());
  }, []);
  useEffect(() => {
    const mediaQueryList = window.matchMedia("(max-width: 768px)");
    mediaQueryList.addListener(handleMediaQuery);
    return () => {
      mediaQueryList.removeListener(handleMediaQuery);
    };
  }, []);
  return (
    <section>
      {isMobile ? (
        <MobileFeedbacks comments={shopComments} />
      ) : (
        <DesktopFeedbacks comments={shopComments} />
      )}
    </section>
  );
};

export default Feedbacks;
