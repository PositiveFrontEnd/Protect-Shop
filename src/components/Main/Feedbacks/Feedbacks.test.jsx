import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import Feedbacks from "./Feedbacks";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
describe("Тестування компонента  Feedbacks", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {
            home: {
                isAnimation: false,
                inputText: "",
                filterHomePage: [],
                marker: null,
            },
            user: {
                token: "token",
                registrationModal: false,
                authorizationStatus: null,
                registrationStatus: "some text",
                registrationData: {},
                correctData: "some text",
                isAdmin: false,
            },
            products: {
                products: [],
                productsByCategory: [],
                productsByType: [],
                oneProduct: {},
                ThreeProducts: [],
                ThreeColors: [],
                filters: [],
                twelveProducts: [],
                filtersParameters: {},
                twelveFiltersProducts: [],
                searchProducts: [],
                youSee: [],
                previewProductInfo: {},
            },
            order: {
                orderState: [],
                productsForOrderGuest: {},
                infoForOrderGuest: [],
                deliveryForOrderGuest: {},
                order: {
                    letterSubject: "Thank you for order!",
                },
                promoCodePrice: "",
                orderGuest: { letterSubject: "Thank you for order!" },
            },
            basket: {
                basket: [],
                priceGuest: "''",
                guestBasket: {},
            },
            catalogue: {
                sortCatalogue: "some text",
                loadFilter: false,
            },
            favorite: {
                favorite: [],
                favoriteForCustomer: [],
                guestFavorite: [],
            },
            message: {
                letters: [],
                letterAll: [],
                importantLetters: []
            },
            shopComments: {
                createShopComment: [],
                getShopComments: [],
            },
            comments: {
                productComments: [],
                productComment: {},

            }
        };

        const store = mockStore(initialState);
        const feedbacks = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Feedbacks />
                </BrowserRouter>
            </Provider>
        );

        expect(feedbacks.container.innerHTML).toMatchSnapshot();
    });
});
