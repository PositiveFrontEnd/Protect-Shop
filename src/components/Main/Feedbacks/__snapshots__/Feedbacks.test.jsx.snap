// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  Feedbacks > Знімок 1`] = `"<section><div class="feedbacks__wrapper"><div class="container feedbacks"><h2 class="feedbacks__title">feedbacks</h2><div class="feedbacks__content"></div><div class="feedbacks__button"><button class="button-white" type="button">see more</button></div></div></div></section>"`;
